### PR TITLE
Support CentOS Stream 9 and RHEL 9

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

File: 01_install_requirements.sh [hunk 1]
```diff
@@ -34,31 +34,50 @@ sudo dnf -y upgrade
...
-elif [[ $DISTRO == "rhel8" ]]; then
-    # Enable EPEL for python3-passlib and python3-bcrypt required by metal3-dev-env
-    sudo dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
-
-    if sudo subscription-manager repos --list-enabled 2>&1 | grep "ansible-2-for-rhel-8-x86_64-rpms"; then
-      # The packaged 2.x ansible is too old for compatibility with metal3-dev-env
-      sudo dnf erase -y ansible
-      sudo subscription-manager repos --disable=ansible-2-for-rhel-8-x86_64-rpms
-    fi
-fi
```
Comment (hardys) [inline 01_install_requirements.sh:49]: RHEL8 still works fine for me and RHEL9 isn't yet released, so I wonder if we can maintain RHEL8 for a bit longer?

**Comment by elfosardo** [comment]
we may need https://github.com/openshift-metal3/dev-scripts/pull/1407 first

**Comment by elfosardo** [comment]
also this is needed https://github.com/openshift/installer/pull/6062

**Comment by elfosardo** [comment]
this passes on current CI platform plus tested on CS8/CS9, it should be enough, just need to wait for the two mentioned PRs and of course reviews

**Comment by elfosardo** [comment]
/retest
this was working fine and it works fine locally, let's give it another try

File: 01_install_requirements.sh [hunk 2]
```diff
@@ -34,41 +34,60 @@ sudo dnf -y upgrade
...
 # vm-setup/install-package-playbook.yml
-# Note recent ansible needs python >= 3.8 so we install 3.9 here
-sudo dnf -y install python39
-sudo alternatives --set python /usr/bin/python3.9
-sudo alternatives --set python3 /usr/bin/python3.9
-sudo update-alternatives --install /usr/bin/pip3 pip3 /usr/bin/pip3.9 1
-sudo pip3 install ansible=="${ANSIBLE_VERSION}"
-# Also need the 3.9 version of netaddr for ansible.netcommon,
+case $DISTRO in
+  "centos8"|"rhel8")
```
Comment (dtantsur) [inline 01_install_requirements.sh:49]: Maybe short-circuit centos8 here to avoid checking for it 4 lines below?

File: 01_install_requirements.sh [hunk 3]
```diff
@@ -34,41 +34,60 @@ sudo dnf -y upgrade
...
+    # install network-scripts package to be able to use legacy network commands
+    sudo dnf install -y network-scripts
+    if [[ $DISTRO == "centos8" ]]; then
+      echo "CentOS is not supported anymore. Please switch to CentOS Stream / RHEL / Rocky Linux"
```
Comment (dtantsur) [inline 01_install_requirements.sh:53]: Fail here?

File: 01_install_requirements.sh (hunk 2 above)
Comment (elfosardo) [reply 01_install_requirements.sh:49]: I will do both changes in a follow up, would rather avoid another CI pass

**Comment by flavio-fernandes** [comment]
hi @elfosardo . I'm trying dev-scripts on centos9-stream and getting the following error:

```
# dnf group install -y --with-optional "Virtualization Host" && systemctl enable --now libvirtd && echo ok

...
level=debug msg=Terraform has been successfully initialized!
level=debug msg=[INFO] running Terraform command: /home/ff/dev-scripts/ocp/ostest/terraform/bin/terraform apply -no-color -auto-approve -input=false -var-file=/tmp/openshift-install-bootstrap-1219846892/terraform.tfvars.json -var-file=/tmp/openshift-install-bootstrap-1219846892/terraform.platform.auto.tfvars.json -lock=true -parallelism=10 -refresh=true
level=error
level=error msg=Error: failed to dial libvirt: dial unix /var/run/libvirt/libvirt-sock: connect: no such file or directory
level=error
level=error msg=  with provider["openshift/local/libvirt"],
level=error msg=  on main.tf line 1, in provider "libvirt":
level=error msg=   1: provider "libvirt" {
level=error
level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failure applying terraform for "bootstrap" stage: failed to create cluster: failed to apply Terraform: exit status 1
level=error
level=error msg=Error: failed to dial libvirt: dial unix /var/run/libvirt/libvirt-sock: connect: no such file or directory
level=error
level=error msg=  with provider["openshift/local/libvirt"],
level=error msg=  on main.tf line 1, in provider "libvirt":
level=error msg=   1: provider "libvirt" {
level=error
level=error
+(utils.sh:1): create_cluster(): auth_template_and_removetmp
+(utils.sh:672): auth_template_and_removetmp(): echo 4
+(utils.sh:673): auth_template_and_removetmp(): generate_auth_template
+(utils.sh:297): generate_auth_template(): set +x
Unable to connect to the server: dial tcp 192.168.111.5:6443: connect: no route to host
make: *** [Makefile:54: ocp_run] Error 1
```

Looking closely at my system, I can see that only the `-ro` file exist:

```
[ff@sdn-12 dev-scripts]$ sudo ls /var/run/libvirt/libvirt-soc*
/var/run/libvirt/libvirt-sock-ro

[root@sdn-12 ~]# virsh list --all
 Id   Name              State
----------------------------------
 -    ostest_master_0   shut off
 -    ostest_master_1   shut off
 -    ostest_master_2   shut off
 -    ostest_worker_0   shut off
 -    ostest_worker_1   shut off

[ff@sdn-12 dev-scripts]$ sudo systemctl status libvirtd
○ libvirtd.service - Virtualization daemon
     Loaded: loaded (/usr/lib/systemd/system/libvirtd.service; enabled; preset: disabled)
     Active: inactive (dead) since Wed 2023-01-11 15:27:44 EST; 8min ago
   Duration: 2min 109ms
TriggeredBy: ○ libvirtd-ro.socket
             ● libvirtd.socket
             ● libvirtd-admin.socket
             ○ libvirtd-tcp.socket
             ○ libvirtd-tls.socket
       Docs: man:libvirtd(8)
             https://libvirt.org
   Main PID: 10358 (code=exited, status=0/SUCCESS)
      Tasks: 2 (limit: 32768)
     Memory: 29.9M
        CPU: 1.246s
     CGroup: /system.slice/libvirtd.service
             ├─4691 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/libexec/libvirt_leaseshelper
             └─4692 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/libexec/libvirt_leaseshelper
```

Can you offer some pointers on whether this is an issue with terraform not using the `-ro` file or if I'm mssinf something that would make `/var/run/libvirt/libvirt-sock` available?

**Comment by flavio-fernandes** [comment]
Update: I managed to work around this issue by **_not_** instaling libvirt explicitly!
Letting dev-scripts install libvirt for me must be different enough, such that `/var/run/libvirt/libvirt-sock` is created /out problems. woot!

**Pull request merged**